### ISSUE TITLE
Property "use-config-as-code-from-default-branch" disallows overriding properties with branch's own cx.config

### ISSUE THREAD

**Issue by cx-scord** [checkmarx-ltd/cx-flow#793]
### Description

Setting `use-config-as-code-from-default-branch` property to `true` will void any possibility of a push/PR on a non-default protected branch overriding some CxFlow properties using the branch's _cx.config_ file. For example, if the default branch's _cx.config_ is filtering Highs and Mediums, by changing it on the protected branch's _cx.config_ file to filter only Highs when the code is pushed is currently impossible.
`
### Expected Behavior

Use the protected branch's _cx.config_ file to override the CxFlow configuration property on a push/pr event.

Could be accomplished by having a non-binary way to tell CxFlow what events should trigger the fetch of config-as-code's _cx.config_ from the default branch, by replacing the `use-config-as-code-from-default-branch: true|false` property with something like:
```yaml
github:
  use-config-as-code-from-default-branch-on:
  - delete
  - pushreq
  - pullreq
```

The default option here should be exclusively for the "delete" event since it's the only event when you usually require a _cx.config_ from the default branch to be fetc  since the branch's _cx.config_ code will not be present after the branch is deleted.

This should be the rule for other supported SCMs as well, like when using Azure DevOps' CxFlow configuration property `deleteCxProject: true|false` that deletes a project when the feature branch is deleted. This should also be changed to something like:

```yaml
azure:
  deleteCxProject-on:
  - delete
  - pushreq
  - pullreq
```
There might be times when using the config-as-code from the default branch might make sense, hence adding the "pushreq" and "pullreq" options

### Actual Behavior

As per the request done on #384, the behavior enforces the pick up of _cx.config_ from the default branch every time (push/PR/delete) and not just on deletion as most cases will be based on.
This means that, for example, if one pushes code or opens a PR on a branch that requires a different _cx.config_, for example, to set a filter to catch only Highs and send them to a bug-tracker, then that will not work and the `filter-severity`

### Reproduction

1) Execute CxFlow in web mode but add the `use-config-as-code-from-default-branch: true` option on the GitHub section.
2) Set a _cx.config_ with overriding properties on a feature branch and push the branch to one of the protected branches,
3) Check if the overrides weren't applied to the  project.

### Environment Details

Windows 10, CxFlow 1,6,22
